v03_pipeline/lib/reference_datasets/clinvar.py

```diff
@@ -0,0 +1,189 @@
+        if not line:
+            continue
+        if line.startswith('##fileDate='):
+            return line.split('=')[-1].strip()
+        if not line.startswith('#'):
+            return None
+    return None
+
+
+def get_submission_summary_ht() -> hl.Table:
+    with tempfile.NamedTemporaryFile(
+        suffix='.txt.gz',
+        delete=False,
+    ) as tmp_file, requests.get(
+        CLINVAR_SUBMISSION_SUMMARY_URL,
+        stream=True,
+        timeout=10,
+    ) as r:
+        shutil.copyfileobj(r.raw, tmp_file)
+    ht = hl.import_table(
+        tmp_file.name,
+        force=True,
+        filter='^(#[^:]*:|^##).*$',  # removes all comments except for the header line
+        types={
+            '#VariationID': hl.tstr,
+            'Submitter': hl.tstr,
+            'ReportedPhenotypeInfo': hl.tstr,
+        },
+        missing='-',
+    )
+    ht = ht.rename({'#VariationID': 'VariationID'})
+    ht = ht.select('VariationID', 'Submitter', 'ReportedPhenotypeInfo')
+    return ht.group_by('VariationID').aggregate(
+        Submitters=hl.agg.collect(ht.Submitter),
+        Conditions=hl.agg.collect(ht.ReportedPhenotypeInfo),
+    )
+
+
+def select_fields(ht):
+    clnsigs = parsed_clnsig(ht)
+    ht = ht.select(
+        alleleId=ht.info.ALLELEID,
+        pathogenicity=hl.if_else(
+            CLINVAR_PATHOGENICITIES_LOOKUP.contains(clnsigs[0]),
+            clnsigs[0],
+            CLINVAR_DEFAULT_PATHOGENICITY,
+        ),
+        assertions=hl.if_else(
+            CLINVAR_PATHOGENICITIES_LOOKUP.contains(clnsigs[0]),
+            clnsigs[1:],
+            clnsigs,
+        ),
+        # NB: there's a hidden enum-mapping inside this clinvar function.
+        conflictingPathogenicities=parsed_and_mapped_clnsigconf(ht),
+        goldStars=CLINVAR_GOLD_STARS_LOOKUP.get(hl.delimit(ht.info.CLNREVSTAT)),
+        submitters=ht.submitters,
+        # assumes the format 'MedGen#:condition', e.g.'C0023264:Leigh syndrome'
+        conditions=hl.map(
+            lambda p: p.split(r':')[1],
+            ht.conditions,
+        ),
+    )
+    return ht.transmute(
+        pathogenicity_id=enum_map(ht.pathogenicity, CLINVAR_PATHOGENICITIES),
+        assertion_ids=enum_map(ht.assertions, CLINVAR_ASSERTIONS),
+    )
+
+
+def get_ht(
+    clinvar_url: str,
+    reference_genome: ReferenceGenome,
+    dataset_type: DatasetType,
+) -> hl.Table:
+    version = parse_clinvar_release_date(clinvar_url)
+    with tempfile.NamedTemporaryFile(
+        suffix='.vcf.gz',
+        delete=False,
+    ) as tmp_file, requests.get(clinvar_url, stream=True, timeout=10) as r:
+        shutil.copyfileobj(r.raw, tmp_file)
+    ht = hl.import_vcf(
+        tmp_file.name,
+        reference_genome=reference_genome.value,
+        drop_samples=True,
+        skip_invalid_loci=True,
+        contig_recoding=reference_genome.contig_recoding(include_mt=True),
+        force_bgz=True,
+    ).rows()
+    ht = filter_contigs(ht, reference_genome, dataset_type)
```

Comment:
I thought we agreed to maintain one table per dataset type and do the filtering at join time

Reply:
Yes, that's correct! I did this wrong.

Reply:
I removed `dataset_type` from this function and we'll manage the filtering to `mito` downstream when we annotate the annotations table, which was my original thought and got ahead of myself here.